bundles/binding/org.openhab.binding.sonance/src/main/java/org/openhab/binding/sonance/internal/SonanceBinding.java

```diff
@@ -0,0 +1,447 @@
+
+import org.openhab.binding.sonance.SonanceBindingProvider;
+
+import org.apache.commons.lang.StringUtils;
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.binding.BindingProvider;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.IncreaseDecreaseType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implement this class if you are going create an actively polling service like
+ * querying a Website/Device.
+ * 
+ * @author Laurens Van Acker
+ * @since 1.8.0
+ */
+public class SonanceBinding extends
+		AbstractActiveBinding<SonanceBindingProvider> {
+	private Map<String, Socket> socketCache = new HashMap<String, Socket>();
+	private Map<String, DataOutputStream> outputStreamCache = new HashMap<String, DataOutputStream>();
+	private Map<String, BufferedReader> bufferedReaderCache = new HashMap<String, BufferedReader>();
+	private Map<String, ReentrantLock> volumeLocks = new HashMap<String, ReentrantLock>();
+
+	private static final Logger logger = LoggerFactory
+			.getLogger(SonanceBinding.class);
+
+	/**
+	 * the refresh interval which is used to poll values from the Sonance server
+	 * (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 60000;
+
+	public SonanceBinding() {
+	}
+
+	/**
+	 * Called by the SCR to activate the component with its configuration read
+	 * from CAS
+	 * 
+	 * @param bundleContext
+	 *            BundleContext of the Bundle that defines this component
+	 * @param configuration
+	 *            Configuration properties for this component obtained from the
+	 *            ConfigAdmin service
+	 */
+	public void activate(final BundleContext bundleContext,
+			final Map<String, Object> configuration) {
+		// the configuration is guaranteed not to be null, because the component
+		// definition has the
+		// configuration-policy set to require. If set to 'optional' then the
+		// configuration may be null
+
+		// to override the default refresh interval one has to add a
+		// parameter to openhab.cfg like <bindingName>:refresh=<intervalInMs>
+		String refreshIntervalString = (String) configuration.get("refresh");
+		if (StringUtils.isNotBlank(refreshIntervalString))
+			refreshInterval = Long.parseLong(refreshIntervalString);
+
+		setProperlyConfigured(true);
+	}
+
+	/**
+	 * Called by the SCR when the configuration of a binding has been changed
+	 * through the ConfigAdmin service.
+	 * 
+	 * @param configuration
+	 *            Updated configuration properties
+	 */
+	public void modified(final Map<String, Object> configuration) {
+		String refreshIntervalString = (String) configuration.get("refresh");
+		if (StringUtils.isNotBlank(refreshIntervalString))
+			refreshInterval = Long.parseLong(refreshIntervalString);
```

Comment:
since the AbstractActiveService get initialised and started with the refresh interval changing this member won't be effective until the next start of the binding. So either remove this modified option or restart the AbstractActiveService properly.